src/pykiso/test_coordinator/test_execution.py

```diff
@@ -128,6 +131,24 @@ def set_skipped(test_case: BasicTest) -> None:
     list(map(set_skipped, filter(is_skip_test, base_suite)))
 
 
+def apply_tc_name_filter(
+    all_tests_to_run: unittest.TestSuite, match_expr: str
+) -> unittest.TestSuite:
+    """Apply a filter to run only test cases which matches given expression
+
+    :param all_tests_to_run: a dict containing all testsuites and testcases
+    :param match_expr: matching regex to activate a test case
+    :return: new test suite with filtered test cases
+    """
+
+    base_suite = test_suite.flatten(all_tests_to_run)
+    is_active_tc: Callable[[BasicTest], bool] = lambda test_case: re.match(
+        match_expr, test_case._testMethodName
```

Comment:
Do we want to match the test method name only or always the actual TestCase's name?
(so `test.__class__.__name__`)

Reply:
Currently it works perfect to select test cases inside a testclass

class TestSerialLoopback(pykiso.BasicTest):
    """In this test we try to receive the same message which we have send.
    RX is connected to TX on a serial dongle.
    """

    def setUp(self):
        logging.info('Do setup')

     def test_run1(self):

        pass

    def test_run2(self):
        pass

    def test_run3(self):
        pass

    def test_run4(self):
        pass

    def test_run5(self):
        pass

    def tearDown(self):
        """If a fixture is not use just override it like below."""
        logging.info('Do teardown')
       


If i select -t test_run3  
it runs the 
setup
test3
teardown

if i run  -t no_match
nothing will be executed


test.__class__.__name__ 
returns the testclass name. here 
TestSerialLoopback-1-1

I think it makes no sense to filter at this level